use-cases/using-SPDK-vhostuser-and-kata.md

```diff
@@ -0,0 +1,221 @@
+
+The Storage Performance Development Kit (SPDK) provides a set of tools and
+libraries for writing high performance, scalable, user-mode storage applications.
+
+virtio, vhost and vhost-user:
+- virtio is an efficient way to transport data for virtual environments and
+guests. It is most commonly used in QEMU VMs, where the VM itself exposes a
+virtual PCI device and the guest OS communicates with it using a specific virtio
+PCI driver. Its diagram is:
+```
++---------+------+--------+----------+--+
+|         +------+        +----------+  |
+| user    |      |        |          |  |
+| space   |      |        |  guest   |  |
+|         |      |        |          |  |
+|    +----+ qemu |        | +-+------+  |
+|    |    |      |        | | virtio |  |
+|    |    |      |        | | driver |  |
+|    |    +------+        +-+---++---+  |
+|    |                          |       |
+|    |       ^                  |       |
+|    v       |                  v       |
+|            |                          |
++-+------+----------------+--+-------+--+
+| |block |   +------------+ kvm.ko   |  |
+| |device|                |          |  |
+| +------+                +--+-------+  |
+|                kernel                 |
++---------------------------------------+
+```
+
+- vhost is a protocol for devices accessible via inter-process communication. It
+uses the same virtio queue layout as virtio to allow vhost devices to be mapped
+directly to virtio devices. The initial vhost implementation is a part of the
+Linux kernel and uses an ioctl interface to communicate with userspace
+applications. Its diagram is:
+```
++---------+------+--------+----------+--+
+|         +------+        +----------+  |
+| user    |      |        |          |  |
+| space   |      |        |  guest   |  |
+|         |      |        |          |  |
+|         | qemu |        | +-+------+  |
+|         |      |        | | virtio |  |
+|         |      |        | | driver |  |
+|         +------+        +-+---++---+  |
+|                               |       |
+|                               |       |
+|                               v       |
+|                                       |
++-+------+--+------+------+--+--v-------+
+| |block |  |vhost-scsi.ko|  | kvm.ko   |
+| |device|  |             |  |          |
+| +---^--+  +-v---------^-+  +--v-------+
+|     +-------+  kernel +-------+       |
++---------------------------------------+
+```
+
+- vhost-user implements the control plane through Unix domain socket to establish
+virtio queue sharing with a user space process on the same host. SPDK exposes
+vhost devices via the vhost-user protocol. Its diagram is:
+```
++---------------+------+--+----------+--+
+|               +------+  +----------+  |
+| user          |      |  |          |  |
+| space         |      |  |  guest   |  |
+|               |      |  |          |  |
+|  +-+-------+  | qemu |  | +-+------+  |
+|  | vhost   |  |      |  | | virtio |  |
```

Comment:
is this vhost-backed connected to virtio-driver?

Reply:
Ack.
Add blocks for unix socket and shared memory